### PR TITLE
Restrict Java toolchain download to secure URIs

### PR DESCRIPTION
**Pull request by ljacomet**

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

File: subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/install/internal/AdoptOpenJdkDownloader.java [hunk 1]
```diff
@@ -72,8 +74,17 @@ private void copyIntoFile(URI source, InputStream inputStream, File destination)
...
+    private RepositoryTransport getTransport(URI source) {
+        final HttpRedirectVerifier redirectVerifier;
+        try {
+            redirectVerifier = HttpRedirectVerifierFactory.create(new URI(source.getScheme(), source.getAuthority(), null, null, null), false, () -> {
+                throw new InvalidUserCodeException("Attempting to download a JDK from an insecure URI " + source + ". This is not supported, use a secure URI instead.");
+            }, uri -> {
+                throw new InvalidUserCodeException("Attempting to download a JDK from an insecure URI " + uri + ". This URI was reached as a redirect from " + source + ". This is not supported, make sure no insecure URIs appear in the redirect");
+            });
+        } catch (URISyntaxException e) {
+            throw new InvalidUserCodeException("Cannot extract host information from specified URI " + source);
```
Comment (JLLeitschuh) [inline subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/install/internal/AdoptOpenJdkDownloader.java:86]: Do you intentionally drop the cause here?

**Review by JLLeitschuh — APPROVED** [review]
LGTM!

**Review by melix — APPROVED** [review]

File: subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/install/internal/AdoptOpenJdkDownloader.java (hunk 1 above)
Comment (ljacomet) [reply subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/install/internal/AdoptOpenJdkDownloader.java:86]: This actually cannot happen AFAICT and is made mandatory by the exception not being a runtime one.

**Pull request merged**